launchflow/exceptions.py

```diff
@@ -148,6 +148,13 @@ def __init__(self, service: Any, existing_product: str, new_product: str) -> Non
         )
 
 
+class ServiceMissingDeploymentId(Exception):
+    def __init__(self, service_name: str) -> None:
+        super().__init__(
+            f"Service '{service_name}' is missing a deployment id. Please run `lf deploy {service_name}` to deploy it before promoting."
```

Comment:
nit: I think this error message might be confusing if the user runs into it, they probably don't know what a deployment ID is.

Maybe just: `Service {service_name} hasn't been deployed yet. Please run ...`

Reply:
done